cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,182 @@
+}
+
+// updateApplicationAPI defines a subset of the application facade, as required
+// by the update-series command.
+type updateSeriesAPI interface {
+	Get(string) (*params.ApplicationGetResults, error)
+	GetCharmURL(string) (*charm.URL, error)
+	UpdateApplicationSeries(string, string, bool) error
+}
+
+// charmAPI defines a subset of the charm facade, as required
+// by the update-series command.
+type charmAPI interface {
+	CharmInfo(charmURL string) (*charms.CharmInfo, error)
+}
+
+// updateSeriesCommand is responsible for updating the series of an application or machine.
+type updateSeriesCommand struct {
+	modelcmd.ModelCommandBase
+
+	apiRoot            api.Connection
+	charmClient        charmAPI
+	updateSeriesClient updateSeriesAPI
+
+	applicationName string
+	force           bool
+	machineNumber   string
+	series          string
+}
+
+var updateSeriesDoc = `
+It is recommended to update the series on existing units before running this
+command to update a machines' series.
+
+By updating the application series, future units of the application to use
+the new series.
+
+The update is disallowed unless the --force flag is used if the requested
+series is not explicitly supported by the application's charm and any
+subordinates, as well as any other charms which may be deployed to the same
+machine.
+
+The --force option should be used with caution since using a charm on a machine
+running an unsupported series may cause unexpected behavior.  Alternately, if
+the requested series supported in later revisions of the charm, upgrade-charm
+can run beforehand.
+
+Updating the
+
+Examples:
+	juju update-series <application> <series>
+	juju update-series <application> <series> --force
+	juju update-series <machine> <series>
+	juju update-series <machine> <series> --force
+
+See also:
+    machines
+    status
+    upgrade-charm
+`
+
+func (c *updateSeriesCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-series",
+		Args:    "[<application>|<machine>] <series>",
+		Purpose: "Update an application or machine's series.",
+		Doc:     updateSeriesDoc,
+	}
+}
+
+func (c *updateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.force, "force", false, "Update even if the series is not supported by the charm and/or related subordinate charms.")
+}
+
+// Init implements cmd.Command.
+func (c *updateSeriesCommand) Init(args []string) error {
+	switch len(args) {
+	case 2:
+		if names.IsValidApplication(args[0]) {
+			c.applicationName = args[0]
+		} else if names.IsValidMachine(args[0]) {
+			c.machineNumber = args[0]
+		} else {
+			return errors.Errorf("invalid application name or machine number %q", args[0])
+		}
+		c.series = args[1]
+	case 1, 0:
+		return errors.Errorf("No application/machine and/or series specified")
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+	return nil
+}
+
+// Run connects to the specified environment and starts the application or machine
```

Comment:
We don't use the word "environment" anymore. And there's no process to start - the cmd just does a db update.
The comment in other places is typically "Run implements cmd.Run"